test/integration_test/snapshot_test.go

```diff
@@ -1571,30 +1582,138 @@ func validateSnapshotCleanup(scheduleName, namespace string) (err error) {
 }
 
 func storageclassTests(t *testing.T) {
-	ctxs, err := schedulerDriver.Schedule("autosnaptest",
-		scheduler.ScheduleOptions{AppKeys: []string{"snapshot-storageclass"}})
-	log.FailOnError(t, err, "Error scheduling task")
-	Dash.VerifyFatal(t, len(ctxs), 1, "Only one task should have started")
+	var testrailID, testResult = 50798, testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult, t.Name())
+	defer updateTestRail(&testResult, testrailID, runID)
+	defer updateDashStats(t.Name(), &testResult)
+
+	namespace := "interval-autosnapsched-test"
+	ns := v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+	_, err := core.Instance().CreateNamespace(&ns)
+	if err != nil && !k8s_errors.IsAlreadyExists(err) {
+		log.FailOnError(t, err, "Error creating namespace")
+	}
+	defer func() {
+		err = core.Instance().DeleteNamespace(namespace)
+		log.FailOnError(t, err, fmt.Sprintf("Namespace %v not deleted", namespace))
+	}()
+
+	///////////////////////////////
+	// Create and deploy policy //
+	/////////////////////////////
+	policyName := "interval"
+	policyInterval := 2
+	schedPolicy := &storkv1.SchedulePolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: policyName,
+		},
+		Policy: storkv1.SchedulePolicyItem{
+			Interval: &storkv1.IntervalPolicy{
+				IntervalMinutes: policyInterval,
+			},
+		}}
+
+	if authTokenConfigMap != "" {
+		if schedPolicy.Annotations == nil {
+			schedPolicy.Annotations = make(map[string]string)
+		}
+		schedPolicy.Annotations[secrets.SecretNameKey] = operator_util.SecurityPXUserTokenSecretName
+		schedPolicy.Annotations[secrets.SecretNamespaceKey] = os.Getenv(k8sutils.PxNamespaceEnvName)
+	}
+
+	_, err = storkops.Instance().CreateSchedulePolicy(schedPolicy)
+	log.FailOnError(t, err, "Error creating interval schedule policy")
+	defer func() {
+		err = storkops.Instance().DeleteSchedulePolicy(schedPolicy.Name)
+		log.FailOnError(t, err, fmt.Sprintf("Schedule policy %v not deleted", schedPolicy.Name))
+	}()
+	log.InfoD("Created schedulepolicy %v with %v minute interval", policyName, policyInterval)
+
+	/////////////////////////////////////
+	// Create and deploy storage class //
+	/////////////////////////////////////
+	storageClassName := "autosnap-sc"
```

Comment:
Why are we creating this dynamically instead of applying from a spec from integration test?

Reply:
This is added to configure the storageclass and PVC to use the `px-user-token` instead of the admin token which gets automatically configured if we deploy applications using spec in integration tests.